python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -84,6 +84,34 @@ def sorted_like(self, like: Column, /) -> Self:
             is_sorted=like.is_sorted, order=like.order, null_order=like.null_order
         )
 
+    # TODO: Return Column once #16272 is fixed.
+    def astype(self, dtype: plc.DataType) -> plc.Column:
+        """
+        Return the backing column as the requested dtype.
+
+        Parameters
+        ----------
+        dtype
+            Datatype to cast to.
+
+        Returns
+        -------
+        Column of requested type.
+
+        Raises
+        ------
+        RuntimeError
+            If the cast is unsupported.
+
+        Notes
+        -----
+        This only produces a copy if the requested dtype doesn't match
+        the current one.
+        """
+        if self.obj.type() != dtype:
+            return plc.unary.cast(self.obj, dtype)
```

Comment:
I think there's a misunderstanding here. `astype` turns a `Column` with dtype `X` into one with dtype `Y`. If `Y == X`, we don't do anything. If `Y != X` then we must do something.

The _precondition_ for calling `astype` right now (because we haven't built out casting utilities) is that `dtypes.can_cast` returns true. So effectively we have:

```
if can_cast(source, target):
    value = column.astype(target) # always succeeds, maybe copies.
```

So I don't think the condition will change.

Reply:
> The precondition for calling astype right now (because we haven't built out casting utilities) is that dtypes.can_cast returns true.

I didn't realize this until later in the review, hence my confusion. I forgot to come back to this thread. Carry on, nothing to see here.